src/DynamicExpresso.Core/Identifier.cs

```diff
@@ -22,13 +22,13 @@ public Identifier(string name, Expression expression)
 		}
 	}
 
-	internal class FunctionIdentifier : Identifier
+	public class FunctionIdentifier : Identifier
```

Comment:
I'm not sure if making `FunctionIdentifier` public is a good idea.
For 2 main reasons:
1. calling `AddOverload` or other functions that change the class state can alter the state of the overall `Interpreter` in unexpected way. See https://github.com/dynamicexpresso/DynamicExpresso#performance-and-multithreading
2. having a wider public API makes refactoring and future extensions more difficult to avoid breaking changes

If there aren't specific needs I would prefer to leave it `private`.

Reply:
You are right, I also think it is not the best way. I only want to replace an existing function identifier instead of adding a new overload.

Reply:
So you need to use an existing `Interpreter` instance but with a different function? Right?
In this case maybe the best way is to simply create a new `Interpreter`. You can for example create a factory to create your Interpreter instances and in one case use the new function.

What do you think?

Reply:
If the existing `Interpreter` has a lot of functions and variables, I need to re-add them when I create a new `Interpreter` for the new function.  And the new function is not very "different", it has the same signature as the old one. It's just wrongly regarded as overload when calling the `SetFunction`, and then the `Interpreter` will can not parse any expression text with that function name.

Now version, `SetIdentifier`, `SetVariable`, and `SetExpression` all will replace the existing expression in the map when setting the same name, but `SetFunction` not.

Reply:
Ok, this is a good point. I agree that this was probably an error, `SetFunction` should replace the old function and not add an overload.
My suggestion is to align the `SetFunction` behaviour with the previous `SetXYZ` functions, replacing the current value, and add another function called `SetFunctionOverload` that instead add an overload if the function already exist.
Unfortunately this will be a breaking change ... but ok, not a big problem, I will create a new major version.

@metoule  What do you think?

Reply:
Ok, get your point @metoule. As a first step we can add a new `ReplaceFunction` that always replace the function no matter if the same signature exist? So if the client wants to explicitly replace any function with the same name can calls this new function.
What do you think? @sagilio @metoule